src/main/java/io/syndesis/verifier/v1/metadata/SalesforceMetadataAdapter.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.verifier.v1.metadata;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.module.jsonSchema.JsonSchema;
+import com.fasterxml.jackson.module.jsonSchema.types.ObjectSchema;
+import com.fasterxml.jackson.module.jsonSchema.types.SimpleTypeSchema;
+
+import org.apache.camel.component.extension.MetaDataExtension.MetaData;
+import org.springframework.stereotype.Component;
+
+@Component("salesforce-adapter")
+public final class SalesforceMetadataAdapter implements MetadataAdapter {
+
+    private static final String UNIQUE_PROPERTY = "sObjectIdName";
+
+    @Override
+    public Map<String, List<PropertyPair>> apply(final Map<String, Object> properties, final MetaData metaData) {
+        final String scope = (String) metaData.getAttribute("scope");
```

Comment:
iiur the 'scope' comes in over the properties send by the syndesis-rest in the endpoint call, so that the metadata extension know what is requested. Since this verifier endpoint is supposed to be universal, does this mean, that every meta data extensions need to support this 'scope' ?

It would be awesome to have a list how the request from syndesis-rest looks like for the various occasions where the service here is queried.

Reply:
`scope` comes from Camel Salesforce component `MetaDataExtension`, it's specific for Salesforce as the MetaDataExtension returns JSON for metadata about all object types, and JSON Schema for specific object type. Properties are entirely up to the `MetaDataExtension` implementation, perhaps it would be good to add a pre-metadata adapter along with this post-metadata adapter (of course in the same interface), perhaps also in another PR?

Reply:
so much meta ;-) still need to understand what are the actual possible requests coming the different possible steps.

But I could be all wrong, it's late, I'm a bit frustrated about the state for the demo for Friday, and have the blues in general ;-| (not the issue of this PR of course, but still not able to get back to the state we already had)